3scale/porta
# THREESCALE-10752: Fix incorrect rendering of Country field

**What this PR does / why we need it**:

The Country field is rendered incorrectly. Instead of the select with the list of countries it shows like this:
![image](https://github.com/3scale/porta/assets/1270328/4e39a675-89d4-4102-9b04-430a380a99b3)

See notes below for more info.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-10752

**Verification steps** 

Add Country to the field definitions for account.
Go to the developer account edit page.
Verify that the country field is shown properly as a select.

**Special notes for your reviewer**:

This issue was caused by this change: https://github.com/3scale/porta/pull/3626/files#diff-1c6f11328cc8f84710e58208eae22ffe8002317fda3641dae36c98ff52c11a16R8

And in fact, it had also been broken for the "new account" form here: https://github.com/3scale/porta/pull/3622/files#diff-2b0651f188471b8b6b1083d2dc23f9b70b328d1564482f6012a9b17fbffb78edR5
(showing an empty input instead of the select with the list of countries).

This is because `PatternflyFormBuilder` only sets the field as a select (i.e. `:patternfly_select`) for custom field with choices (see [this](https://github.com/3scale/porta/blob/0b2efa96557e619c558e3d3da471022d3cfa6527/app/lib/fields/patternfly_form_builder.rb#L5)). But in the original implementation it was Formtastic that identified that `country` was a `:select` by searching for an association reflection with that name.
